### PR TITLE
validate settings

### LINKED ISSUE #42: validate command line arguments (and settings)
See #40
- [x] base url is a valid URL
- [ ] account id parses to Integer
- [x] collection id parses to Integer
- [x] crawl id (if provided) parses to integer (crawl ID is AKA job ID)
- [x] crawl start after date parses to ISO8601 UTC format (e.g. "2017-04-01T12:01:05Z", and "2017-04-01" parses to "2017-04-01T23:59:59Z")
- [x] crawl start before ditto

### PR DESCRIPTION
**Pull request by jmartin-sul**
closes #42

### CODE REVIEW

**Review by tingulfsen — APPROVED** [review]
Looks good from here 👍

File: config/settings.properties [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # WASAPI URL endpoint
-baseurl=http://example.org
+baseurl=https://example.org
```
Comment (ndushay) [inline config/settings.properties:2]: yay for all these exemplar settings changes!

File: src/edu/stanford/dlss/was/WasapiDownloaderSettings.java [hunk 2]
```diff
@@ -62,15 +68,16 @@ public WasapiDownloaderSettings(String settingsFileLocation, String[] args) thro
...
-      //TODO: validate settings state.  see https://github.com/sul-dlss/wasapi-downloader/issues/42
+      validateSettings();
     } catch (IOException e) {
       throw new SettingsLoadException("Error reading properties file: " + e.getMessage(), e);
     } catch (ParseException e) {
       throw new SettingsLoadException("Error parsing command line arguments: " + e.getMessage(), e);
     }
   }
 
+  protected WasapiDownloaderSettings() { } //empty constructor, for testing
```
Comment (ndushay) [inline src/edu/stanford/dlss/was/WasapiDownloaderSettings.java]: i think `// for testing`  is sufficient  (we can see it's an empty constructor.  ;-)

**Review by ndushay — CHANGES_REQUESTED** [review]
One very small quibble.

File: src/edu/stanford/dlss/was/WasapiDownloaderSettings.java (hunk 2 above)
Comment (jmartin-sul) [reply src/edu/stanford/dlss/was/WasapiDownloaderSettings.java]: ha, yes, classic uninformative comment

**Comment by jmartin-sul** [comment]
fixed the comment @ndushay pointed out, re-pushed

**Review by ndushay — APPROVED** [review]

**Pull request merged**